apps/core/app/[locale]/(default)/login/_components/change-password-form.tsx

```diff
@@ -73,18 +78,46 @@ export const ChangePasswordForm = ({ customerId, customerToken }: Props) => {
     messageText = t('successMessage');
   }
 
-  const handleNewPasswordChange = (e: ChangeEvent<HTMLInputElement>) =>
-    setNewPasssword(e.target.value);
+  const handleCurrentPasswordChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setIsCurrentPasswordValid(!e.target.validity.valueMissing);
+  const handleNewPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    let currentPasswordValue: FormDataEntryValue | null = null;
+    let isValid = true;
+    const newPasswordValue = e.target.value;
+
+    if (e.target.form) {
+      currentPasswordValue = new FormData(e.target.form).get('current-password');
+    }
+
+    if (isLoggedIn) {
+      isValid = !e.target.validity.valueMissing && newPasswordValue !== currentPasswordValue;
+    } else {
+      isValid = !e.target.validity.valueMissing;
+    }
+
+    setIsNewPasswordValid(isValid);
+  };
   const handleConfirmPasswordValidation = (e: ChangeEvent<HTMLInputElement>) => {
-    const confirmPassword = e.target.value;
+    let newPasswordValue: FormDataEntryValue | null = null;
+    const confirmPasswordValue = e.target.value;
 
-    return setIsConfirmPasswordValid(confirmPassword === newPassword);
+    if (e.target.form) {
+      newPasswordValue = new FormData(e.target.form).get('new-password');
+    }
+
+    setIsConfirmPasswordValid(
+      confirmPasswordValue.length > 0 && newPasswordValue === confirmPasswordValue,
+    );
   };
```

Comment:
Can we use the schema `CustomerChangePasswordSchema` here? We can modify it to check for equality and length requirements.

Reply:
I've refactored this part but not sure if it's for good. it looks more bulky now. What do you say?